packages/core/src/components/misc/CredentialCard10.tsx

```diff
@@ -143,23 +146,32 @@ const CredentialCard10: React.FC<CredentialCard10Props> = ({ credential, style =
       return
     }
 
-    const params = {
-      identifiers: getCredentialIdentifiers(credential),
-      attributes: buildFieldsFromAnonCredsCredential(credential),
-      meta: {
-        credConnectionId: credential?.connectionId,
-        alias: credentialConnectionLabel,
-      },
-      language: i18n.language,
-    }
-    bundleResolver.resolveAllBundles(params).then((bundle) => {
+    (async () => {
+      const identifiers = getCredentialIdentifiers(credential)
+      const attributes = buildFieldsFromAnonCredsCredential(credential)
+      const credName = await getCredentialName(
+        identifiers.credentialDefinitionId,
+        identifiers.schemaId,
+        agent
+      )
```

Comment:
suggestion: looks like a similar pattern is used a few times throughout this PR, might be a good place to write a custom hook ie: `useCredential` or similarly named where we can load and return these attributes => `{ identifiers, attributes, name }`